src/SimplePie.php

```diff
@@ -804,6 +812,14 @@ public function set_raw_data(string $data)
      */
     public function set_timeout(int $timeout = 10)
     {
+        if ($this->http_client_injected) {
+            throw new SimplePieException(sprintf(
+                'Using "%s()" has no effect, because you already provided a HTTP client with "%s::set_http_client()". Configure timeout in your HTTP client instead.',
+                __METHOD__,
+                self::class
+            ));
+        }
+
```

Comment:
We should unset a possible created $this->http_client on calling` set_timeout()`, `set_curl_options()`, etc.
```suggestion

        // Reset a possible existing FileClient,
        // so a new client with the changed value will be created
        $this->http_client = null;
```

Reply:
The code will throw an exception so the value should not be changed, though.

Reply:
Oh, I see... 🤦